wallet/lib/pages/detailed_asset_view/widgets/wallpaper_set_screen.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:pylons_wallet/components/buttons/custom_paint_button.dart';
+import 'package:pylons_wallet/pages/detailed_asset_view/widgets/nft_image_asset.dart';
+import 'package:pylons_wallet/utils/constants.dart';
+import 'package:pylons_wallet/utils/svg_util.dart';
+import 'package:wallpaper/wallpaper.dart';
+
+
+class WallpaperScreen extends StatefulWidget {
+  const WallpaperScreen({Key? key, required this.nft}) : super(key: key);
+  final String nft;
+
+  @override
+  State<WallpaperScreen> createState() => _WallpaperScreenState();
+}
+
+class _WallpaperScreenState extends State<WallpaperScreen> {
+  ValueNotifier<bool> downloading = ValueNotifier<bool>(false);
+  ValueNotifier<bool> done = ValueNotifier<bool>(false);
+
+  Future<void> downloadAndSetImage(BuildContext context) async {
+    final Stream<String> progressString = Wallpaper.imageDownloadProgress(widget.nft);
+    progressString.listen((data) {
+      downloading.value = true;
+    }, onDone: () async {
+      downloading.value = false;
+      Wallpaper.lockScreen(options: RequestSizeOptions.RESIZE_FIT);
+      done.value = true;
+      //Navigator.pop(context);
+    }, onError: (error) {
+      downloading.value = false;
+    });
+  }
+
+
+  @override
+  Widget build(BuildContext context) {
+    return Material(
```

Comment:
Use colorbox instead of material here.

Reply:
Wasn't able to use ColoredBox since inkwell needs to have a material widget as an ancestor